src/Sylius/Bundle/CoreBundle/Tests/DependencyInjection/CircularDependencyBreakingErrorListenerPassTest.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Tests\DependencyInjection;
+
+use Matthias\SymfonyDependencyInjectionTest\PhpUnit\AbstractCompilerPassTestCase;
+use Sylius\Bundle\CoreBundle\DependencyInjection\Compiler\CircularDependencyBreakingErrorListenerPass;
+use Sylius\Bundle\CoreBundle\EventListener\CircularDependencyBreakingErrorListener;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+final class CircularDependencyBreakingErrorListenerPassTest extends AbstractCompilerPassTestCase
+{
+
+    public function it_register_circular_dependency_breaking_error_listener_when_exception_listener_is_registered(): void
+    {
+        $this->container->setDefinition('exception_listener', new Definition('ExceptionListener'));
+
+        $definition = new Definition(CircularDependencyBreakingErrorListener::class);
```

Comment:
Hmm, we shouldn't create this service here - it should be created by the compiler pass. This way the compiler pass logic isn't really tested.